### PR TITLE
Add `plot_wigner_mosaic`

### PR DESCRIPTION
**Pull request by pierreguilmin**
This would be much better with batched support for wigner, but I don't know how involved it is @gautierronan?

### CODE REVIEW

**Comment by gautierronan** [comment]
Batched support for wigner is probably not so easy. Probably 2 to 4 hours of work. A single wigner is already not super easy to vectorize. There are quite a few for loops lying around.

**Comment by pierreguilmin** [comment]
> Batched support for wigner is probably not so easy. Probably 2 to 4 hours of work. A single wigner is already not super easy to vectorize. There are quite a few for loops lying around.

Ok, no worries then 👍

File: dynamiqs/plots/misc.py [hunk 1]
```diff
@@ -161,6 +170,98 @@ def plot_wigner(
...
+):
+    r"""Plot the Wigner function of multiple states in a mosaic arrangement.
+
+    Warning:
+        Documentation redaction in progress.
+
+    See [`dq.plot_wigner()`][dynamiqs.plot_wigner] for more details.
+
+    Examples:
+        >>> psis = [dq.fock(8, i) for i in range(3)]
```
Comment (gautierronan) [inline dynamiqs/plots/misc.py:196]:
```suggestion
        >>> psis = [dq.fock(3, i) for i in range(3)]
```

File: dynamiqs/plots/misc.py [hunk 2]
```diff
@@ -161,6 +170,98 @@ def plot_wigner(
...
+
+        ![plot_wigner_mosaic_fock](/figs-code/plot_wigner_mosaic_fock.png){.fig}
+
+        >>> n = 16
+        >>> a = dq.destroy(n)
+        >>> H = dq.zero(n)
+        >>> jump_ops = [a @ a - 3.0 * dq.eye(n)]
+        >>> rho0 = dq.coherent_dm(n, 0)
+        >>> tsave = np.linspace(0, 1.0, 101)
+        >>> result = dq.mesolve(H, jump_ops, rho0, tsave)
```
Comment (gautierronan) [inline dynamiqs/plots/misc.py:208]:
```suggestion
        >>> n = 16
        >>> a = dq.destroy(n)
        >>> H = dq.zero(n)
        >>> jump_ops = [a @ a - 4.0 * dq.eye(n)]
        >>> psi0 = dq.coherent(n, 0)
        >>> tsave = np.linspace(0, 1.0, 101)
        >>> result = dq.mesolve(H, jump_ops, psi0, tsave)
```

File: dynamiqs/plots/misc.py [hunk 3]
```diff
@@ -161,6 +170,98 @@ def plot_wigner(
     )
 
 
+def plot_wigner_mosaic(
+    states: ArrayLike,
+    *,
+    n: int = 8,
+    ncols: int | None = None,
```
Comment (gautierronan) [inline dynamiqs/plots/misc.py:177]: I think `nrows` would be more natural. Because if not specified, then `nrows = 1`.

**Review by gautierronan — CHANGES_REQUESTED** [review]

File: dynamiqs/plots/misc.py (hunk 3 above)
Comment (pierreguilmin) [reply dynamiqs/plots/misc.py:177]: Right 👍

**Review by gautierronan — APPROVED** [review]

**Pull request merged**